### PR TITLE
feature/add-create-example-events-functions

### PR DESCRIPTION
**Pull request by tohuynh**
**Pull request recommendations:**
- [x] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [x] Link to any relevant issue in the PR description. Ex: _Resolves [gh-12], adds tiff file format support_
WIP #2 
Related to #3 
- [x] Provide context of changes.
In order to create a `cdp-example` repo with fake data, add functions to generate fake data based on the event [ingestion models](https://github.com/CouncilDataProject/cdp-backend/blob/main/cdp_backend/pipeline/ingestion_models.py) defined in `cdp-backend`. Basically create a function similar to `get_events` that returns a list of `EventIngestionModel` which can then be utilized in the EventGatherPipeline to load into the database.

Further question: Do you think we should just place the content `example_scraper.py` inside `get_events` as "starter code" and add instructions to remove the fake code to provide an actual implementation? In the `cdp-example` repo, we can leave `get_events` as is, so that the EventGatherPipeline will load fake data?
- [ ] Provide relevant tests for your feature or bug fix.
- [ ] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

### CODE REVIEW

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        description="Example Description",
+    )
+    # Create sessions for the event
+    sessions = [
+        Session(
+            session_datetime=datetime.utcnow(),
+            video_uri="https://youtu.be/UjieH-NrOe8",
+        )
+        for _ in range(random.randint(1, 3))
+    ]
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.municipality_slug }}/python/example_scraper.py:51]: For the sake of having some difference in session can you do `datetime.utcnow() + (i * timedelta(hours=3))` where `i` is the enumeration of the random gen list item?

Comment (JacksonMaxfield) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:51]: and maybe do `random.choice()` from a list of options for the video?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    event_minutes_items = None
+    # Create event minutes items for the event
+    if not is_minimal:
+        # Get a number of event minutes items
+        event_minutes_items_num = random.randint(5, 15)
+        # Get a number of sponsors for each event minutes item
+        sponsors_nums = random.choices(range(1, 4), k=event_minutes_items_num)
+        # Specify the seat position for each sponsor
+        sponsors_seats = [
+            random.sample(range(1, FAKE_COUNCIL_SEAT_NUM + 1), sponsors_num)
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.municipality_slug }}/python/example_scraper.py:61]: Nice. I would have forgotten that political seats aren't zero indexed.

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                        name=f"Example Supporting File Name {file_num}",
+                        uri="https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf",
+                    )
+                    for file_num in range(1, random.randint(1, 5) + 1)
+                ],
+                decision=f"Example Decision {random.randint(1, 5)}",
+                votes=[
+                    Vote(
+                        person=_get_example_person(seat_num),
+                        decision=f"Example Decision {random.randint(1, 5)}",
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.municipality_slug }}/python/example_scraper.py:90]: What are the four decision values possible?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                    ],
+                ),
+                supporting_files=[
+                    SupportingFile(
+                        name=f"Example Supporting File Name {file_num}",
+                        uri="https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf",
+                    )
+                    for file_num in range(1, random.randint(1, 5) + 1)
+                ],
+                decision=f"Example Decision {random.randint(1, 5)}",
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.municipality_slug }}/python/example_scraper.py:86]: Hmmmm should we compute this based off individual votes? Would there ever be a situation where that mapping _isn't_ valid?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                ],
+            )
+            for i in range(event_minutes_items_num)
+        ]
+        # Insert a non-matter event minutes item
+        event_minutes_items.insert(
+            0,
+            EventMinutesItem(
+                minutes_item=MinutesItem(
+                    name="Example Minutes Item", description="Example Description"
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.municipality_slug }}/python/example_scraper.py:102]: Maybe make this one like "Approval of Agenda" or something?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        seat=Seat(
+            name=f"Example Seat Position {seat_num}",
+            electoral_area=f"Example Electoral Area {seat_num}",
+            electoral_type=f"Example Electoral Type {1 if seat_num <= FAKE_COUNCIL_SEAT_NUM / 2 else 2 }",
+            image_uri="https://councildataproject.github.io/imgs/seattle.jpg",
+        ),
+        roles=[
+            Role(title="Councilmember", body=Body(name="Example Committee")),
+            Role(title="Chair", body=Body(name=f"Example Committee {seat_num}")),
+        ],
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.municipality_slug }}/python/example_scraper.py:132]: Maybe if the `seat_num` is 1 they should also be council president?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+def get_example_events(
+    n: int = 10, is_minimal: bool = False
+) -> List[EventIngestionModel]:
+    return [_get_example_event(is_minimal) for _ in range(n)]
+
+
+###############################################################################
+
+
+FAKE_COUNCIL_SEAT_NUM = 10
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.municipality_slug }}/python/example_scraper.py:34]: 1. nitpick: can we change this to `NUM_COUNCIL_SEATS`
2.
I would make all the range values constants here as well.

```python
RAND_BODY_RANGE = (1, 100)
RAN_EVENTS_MINUTES_ITEMS_RANGE = (5, 15)
```

etc.

**Review by JacksonMaxfield — CHANGES_REQUESTED** [review]
Strictly speaking about this PR:
1. This is awesome! Most of my comments should be simple ways to improve it further (from my perspective anyway)
2. Can you move the hidden functions defintions to order of usage? It should be `_get_example_person` -> `_get_example_event`

Higher level question. I feel like this should live in the to be made: `councildataproject/example` repo _as_ the `get_events` function. Thoughts?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+    body = Body(
+        name=f"Example Committee {random.randint(1, 100)}",
+        description="Example Description",
+    )
+    # Create sessions for the event
+    sessions = [
+        Session(
+            session_datetime=datetime.utcnow(),
+            video_uri="https://youtu.be/UjieH-NrOe8",
+        )
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.municipality_slug }}/python/example_scraper.py:49]: Can we have a random of 80% of the time the session _has_ a closed caption file?

**Review by JacksonMaxfield — CHANGES_REQUESTED** [review]

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 1 above)
Comment (tohuynh) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:51]: Will do both

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 3 above)
Comment (tohuynh) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:90]: I've seen Did not  vote, abstain, not present. I've seen more than five decision types for sure

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 6 above)
Comment (tohuynh) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:132]: The `seat_num` is based on district areas like District 1 or Citywide position. I think any of the seat positions can be president right?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 4 above)
Comment (JacksonMaxfield) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:86]: Actually. I don't trust the math on our side to compute this. There are definitely special cases where a vote requires a super majority to pass and such and not just a simple majority. Having this field is better than having to bake in all those cases.

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 6 above)
Comment (JacksonMaxfield) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:132]: Totally agree. I was just offering a method to select a council present.

Comment (JacksonMaxfield) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:132]: In a consistent manner between runs that is.

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 4 above)
Comment (tohuynh) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:86]: Are you referring to `decision`? It just picks a random decision on the event minutes item

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 3 above)
Comment (JacksonMaxfield) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:90]: I think our system would ignore "did not vote" and "not present" but "Abstain" is totally valid. We should include these options as constants in the `cdp-backend` library.

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 6 above)
Comment (tohuynh) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:132]: Maybe I can add `Role` council president for Example Person 1?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 4 above)
Comment (JacksonMaxfield) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:86]: Yea. I think this particular case should only have three possible which is "Passed", "Rejected", "Did not address" or something similar.

Comment (JacksonMaxfield) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:86]: I remember Emily and/ or Katlyn did some work in determine all the possible values for matters some where. Maybe pull those in?

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 6 above)
Comment (JacksonMaxfield) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:132]: Yea. Works for me.

File: {{ cookiecutter.municipality_slug }}/python/example_scraper.py (hunk 4 above)
Comment (tohuynh) [reply {{ cookiecutter.municipality_slug }}/python/example_scraper.py:86]: It was Adopted, Rejected, In Progress. I'll change it

**Comment by tohuynh** [comment]
Closing this in favor of [#1](https://github.com/CouncilDataProject/example/pull/1)

**Pull request closed without merging**